extra/level_zero.py

```diff
@@ -0,0 +1,116 @@
+import ctypes
+import tinygrad.runtime.autogen.ze as ze
+from tinygrad.helpers import from_mv, init_c_struct_t
+
+
+def check(status):
+  if status != ze.ZE_RESULT_SUCCESS: raise RuntimeError(f"LEVEL ZERO Error {status}")
+
+check(ze.zeInit(0))
+check(ze.zeDriverGet(ctypes.byref(driver_count := ctypes.c_uint32()), None))
+print(driver_count)
+drivers = (ze.ze_driver_handle_t*driver_count.value)()
+check(ze.zeDriverGet(ctypes.byref(driver_count), drivers))
+print(driver_count, drivers)
+driver = drivers[0]
+
+check(ze.zeDeviceGet(driver, ctypes.byref(device_count := ctypes.c_uint32()), None))
+print(device_count)
+devices = (ze.ze_device_handle_t*device_count.value)()
+check(ze.zeDeviceGet(driver, ctypes.byref(device_count), devices))
+print(device_count, devices)
+
+agents = []
+for dev in devices:
+  device_properties = ze.ze_device_properties_t()
+  device_properties.stype = ze.ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES
+  check(ze.zeDeviceGetProperties(dev, ctypes.byref(device_properties)))
+  if device_properties.type == ze.ZE_DEVICE_TYPE_GPU: agents.append(dev)
+
+print(agents)
+
+context_desc = ze.ze_context_desc_t()
+context_desc.stype = ze.ZE_STRUCTURE_TYPE_CONTEXT_DESC
+check(ze.zeContextCreate(driver, ctypes.byref(context_desc), ctypes.byref(context := ze.ze_context_handle_t())))
+
+altdesc = ze.ze_command_queue_desc_t()
+altdesc.stype = ze.ZE_STRUCTURE_TYPE_COMMAND_QUEUE_DESC
+check(ze.zeCommandListCreateImmediate(context, agents[0], ctypes.byref(altdesc), ctypes.byref(cmdlist := ze.ze_command_list_handle_t())))
+
+# Create an event to be signaled by the device
+ep_desc = ze.ze_event_pool_desc_t()
+ep_desc.stype = ze.ZE_STRUCTURE_TYPE_EVENT_POOL_DESC
+ep_desc.count = 1
+ep_desc.flags = ze.ZE_EVENT_POOL_FLAG_HOST_VISIBLE
+check(ze.zeEventPoolCreate(context, ctypes.byref(ep_desc), 1, ctypes.byref(agents[0]), ctypes.byref(event_pool := ze.ze_event_pool_handle_t())))
+ 
+ev_desc = ze.ze_event_desc_t()
+ev_desc.stype = ze.ZE_STRUCTURE_TYPE_EVENT_DESC
+ev_desc.signal = ze.ZE_EVENT_SCOPE_FLAG_HOST
+ev_desc.wait = ze.ZE_EVENT_SCOPE_FLAG_HOST
+check(ze.zeEventCreate(event_pool, ctypes.byref(ev_desc), ctypes.byref(event := ze.ze_event_handle_t())))
+
+# signal the event from the device and wait for completion
+
+
+print("trying to create module")
+from tinygrad.runtime.ops_gpu import CLCompiler
+from tinygrad import Device
+spriv_bin = CLCompiler(Device["GPU"], "test").compile(f"""
```

Comment:
Just reused OpenCL binary, but I think we need to compile to spirv here. Haven't found any online(rt) OpenCL->spirv compiler yet.

Reply:
Hmm, OpenCL outputs an ELF in the intel GPU assembly, I wouldn't want to put in some intermediate language.